File: development/NET/samples/server/Console/CS/MyAeAddressSpaceElement.cs
using Softing.OPCToolbox;
using Softing.OPCToolbox.Server;

namespace Console
{
	public class MyAeAddressSpaceElement : AeAddressSpaceElement
	{
		#region Constructors
		//-------------------

		public MyAeAddressSpaceElement(
			string aName,
			uint anUserData,
			uint anObjectHandle,
			uint aParentHandle)
			:
			base(
			aName,
			anUserData,
			anObjectHandle,
			aParentHandle)
		{
		}   //	end constructor

		public MyAeAddressSpaceElement()
		{
		}   //	end constructor

		//--
		#endregion

		#region Public Methods
		//-------------------

		public override int QueryConditions(
			string aSourcePath,
			out string[] aConditionNames)
		{

			aConditionNames = new string[2] { OpcServer.COND_NAME_BETWEEN_MULTIPLE, OpcServer.COND_NAME_BETWEEN_SINGLE };
			return (int)EnumResultCode.S_OK;
		}   //	end QueryConditions

		//--
		#endregion

	}   //	end class MyAeAddressSpaceElemen
}

File: development/NET/samples/server/Console/VB/MyAeAddressSpaceElement.vb
Imports System
Imports Softing.OPCToolbox
Imports Softing.OPCToolbox.Server

Namespace Console
	Public Class MyAeAddressSpaceElement
		Inherits AeAddressSpaceElement
#Region "Constructors"
		'-------------------

		Public Sub New(ByVal aName As String, ByVal anUserData As UInt32, ByVal anObjectHandle As UInt32, ByVal aParentHandle As UInt32)
			MyBase.New(aName, anUserData, anObjectHandle, aParentHandle)
		End Sub

		Public Sub New()
		End Sub

		'--
#End Region

#Region "Public Methods"
		'-------------------

		Public Overloads Overrides Function QueryConditions(ByVal aSourcePath As String, ByRef aConditionNames As String()) As Integer

			aConditionNames = New String(1) {OpcServer.COND_NAME_BETWEEN_MULTIPLE, OpcServer.COND_NAME_BETWEEN_SINGLE}
			Return EnumResultCode.S_OK
		End Function

		'--
#End Region

	End Class
End Namespace

